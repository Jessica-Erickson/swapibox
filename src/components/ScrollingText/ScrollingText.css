.ScrollingText {
  grid-column: 1 / 2;
  grid-row: 1 / 3;
  color: gold;
  font-size: 2.5rem;
  text-align: justify;
  overflow: hidden;
  perspective: 400px;
  perspective-origin: 50% 30%;
}

.title {
  line-height: 1;
  margin-bottom: 0;
  text-align: left;
}

.release-date {
  margin-top: 0.5rem;
}

.scrolling-wrapper {
  animation: crawl 40s linear infinite;
  margin: 0 auto;
  position: relative;
  top: -80vh;
  transform: rotateX(55deg) translate3d(0, 885px, 325px);
  width: 80%;
  z-index: 5;
}

.fade {
  position: relative;
  height: 80vh;
  width: 100%;
  background: linear-gradient(black, rgba(0, 0, 0, 0.8), transparent, transparent, transparent, transparent);
  z-index: 10;
}

.display-none {
  display: none;
}

@keyframes crawl {
  from {
    transform: rotateX(55deg) translate3d(0, 885px, 325px);
  }
  to {
    transform: rotateX(55deg) translate3d(0, -3000px, 325px);
  }
}