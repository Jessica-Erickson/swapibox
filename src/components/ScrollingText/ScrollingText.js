import React, { Component } from 'react';
import * as API from '../../helper.js';
import PropTypes from 'prop-types';
import './ScrollingText.css';

class ScrollingText extends Component {
  constructor() {
    super();
    this.state = {
      allFilms: [],
      displayFilm: 0
    }
  }

  componentDidMount = async () => {
    let allFilms;
    if (!localStorge.getItem('allFilms')) {
      allFilms = JSON.parse(localStorge.getItem('allFilms'));
    } else {
      allFilms = await API.films();
      localStorge.setItem('allFilms', JSON.stringify(allFilms));
    }
    this.setState({ allFilms }, this.props.loadingCheck());
  }

  changeText = () => {
    const newFilm = Math.floor(Math.random() * 7);
    this.setState({ film: newFilm });
  }

  render() {
    const { allFilms , displayFilm } = this.state;
    const { display } = this.props;
    const currentFilm = allFilms[currentFilm];
    const { title , releaseDate , openingCrawl } = currentFilm;

    return (
      <aside 
        className={display ? 'ScrollingText' : 'display-none')}>
        <div className='fade'></div>
        <div 
          className={display ? 'scrolling-wrapper' : ''}
          onAnimationIteration={this.changeText}>
          <p className='opening-crawl'>
            {openingCrawl}
          </p>
          <h2 className='title'>
            {title}
          </h2>
          <h3 className='release-date'>
            {releaseDate}
          </h3>
        </div>
      </aside>
    )
  }
}

ScrollingText.propTypes = {
  display: PropTypes.bool.isRequired,
  loadingCheck: PropTypes.func.isRequired
}

export default ScrollingText;