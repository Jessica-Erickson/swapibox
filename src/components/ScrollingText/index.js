import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ScrollingText.css';


class ScrollingText extends Component {
  constructor() {
    super();
    this.state = {
      film: 0
    }
  }

  changeText = () => {
    const newFilm = Math.floor(Math.random() * 7);
    this.setState({ film: newFilm });
  }

  render() {
    const currentFilm = this.props.allFilms[this.state.film]
    const { title, releaseDate, openingCrawl } = currentFilm;

    return (
      <aside
        className="ScrollingText">
        <div className="fade"></div>
        <div 
          className="scrolling-wrapper" 
          onAnimationEnd={this.changeText}>
          <p className="opening-crawl">{openingCrawl}</p>
          <h2 className="title">{title}</h2>
          <h3 className="release-date">{releaseDate}</h3>
        </div>
      </aside>
    )
  }
}

ScrollingText.propTypes = {
  allFilms: PropTypes.array.isRequired
}

export default ScrollingText;