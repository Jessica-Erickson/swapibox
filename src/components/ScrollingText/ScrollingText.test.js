import React from 'react'
import { shallow , mount } from 'enzyme'
import ScrollingText from './index'

describe('ScrollingText', () => {
  let wrapper;
  let allFilms;

  beforeEach(() => {
    allFilms = [
          {title: 'Star Wars 1',
            releaseDate: '1977',
            openingCrawl: 'This is opening crawl 1'},
          {title: 'Star Wars 2',
            releaseDate: '1978',
            openingCrawl: 'This is opening crawl 2'},
          {title: 'Star Wars 3',
            releaseDate: '1979',
            openingCrawl: 'This is opening crawl 3'},
          {title: 'Star Wars 7',
            releaseDate: '1980',
            openingCrawl: 'This is opening crawl 7'}
        ]
    wrapper = shallow(<ScrollingText allFilms={allFilms} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    wrapper = shallow(<ScrollingText allFilms={allFilms} />, { disableLifecycleMethods: true });

    expect(wrapper.state()).toMatchSnapshot();
  });

  it('should know which movie text to show', () => {
    expect(wrapper.state('film')).toEqual(0);
  });

  it('should get a new number between -1 & 8', () => {
    wrapper = mount(<ScrollingText allFilms={allFilms} />);
    
    wrapper.instance().changeText();

    expect(wrapper.state('film')).toBeGreaterThan(-1);
    expect(wrapper.state('film')).toBeLessThan(8);
  });

  it('should set state with a new random number', () => {
    const initialState = 0;
    const expected = 3;

    Math.random = () => { return 0.5 }

    wrapper.setState({ film: initialState });

    wrapper.instance().changeText()

    expect(wrapper.state('film')).toEqual(expected)
  });

  it('should call changeText on animation end', () => {
    wrapper = mount(<ScrollingText allFilms={allFilms} />);

    const spy = spyOn(wrapper.instance(), 'changeText');

    wrapper.instance().forceUpdate();

    wrapper.find('.scrolling-wrapper').simulate('animationend');

    expect(spy).toHaveBeenCalled();
  });
});