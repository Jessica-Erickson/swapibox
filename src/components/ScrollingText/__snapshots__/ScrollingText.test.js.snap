// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should have default state 1`] = `
Object {
  "film": 0,
}
`;

exports[`ScrollingText should match the snapshot 1`] = `
<aside
  className="ScrollingText"
>
  <div
    className="fade"
  />
  <div
    className="scrolling-wrapper"
    onAnimationIteration={[Function]}
  >
    <p
      className="opening-crawl"
    >
      This is opening crawl 1
    </p>
    <h2
      className="title"
    >
      Star Wars 1
    </h2>
    <h3
      className="release-date"
    >
      1977
    </h3>
  </div>
</aside>
`;
